.github/workflows/integrate.yaml

```diff
@@ -0,0 +1,70 @@
+name: CI
+
+on:
+  workflow_call:
+
+jobs:
+  build:
+    name: Build rocks
+    runs-on: ubuntu-22.04
+    outputs:
+      rockcraft-out: ${{ steps.rockcraft.outputs.rock }}
+    strategy:
+      matrix:
+        rock:
+          - jupyter-pytorch-cuda-full
+          - jupyter-pytorch-full
+          - jupyter-tensorflow-cuda-full
+          - jupyter-tensorflow-full
+          - jupyter-scipy
+          - notebook-controller
+          - volumes-web-app
+    steps:
+      # Ideally we'd use self-hosted runners, but this effort is still not stable.
+      # This action will remove unused software (dotnet, haskell, android libs, codeql,
+      # and docker images) from the GH runner, which will liberate around 60 GB of storage
+      # distributed in 40GB for root and around 20 for a mnt point.
+      - name: Maximise GH runner space
+        uses: easimon/maximize-build-space@v7
+        with:
+          root-reserve-mb: 40960
+          remove-dotnet: 'true'
+          remove-haskell: 'true'
+          remove-android: 'true'
+          remove-codeql: 'true'
+          remove-docker-images: 'true'
+
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Build ROCK
+        uses: canonical/craft-actions/rockcraft-pack@main
+        id: rockcraft
+        with:
+          path: ./${{ matrix.rock }}
+
+      - name: Upload rock as artefact
+        uses: actions/upload-artifact@v3
+        with:
+            name: kubeflow-rocks
+            path: ${{ steps.rockcraft.outputs.rock }}
```

Comment:
There is no point in building rocks twice (here in this workflow, and then [here](https://github.com/canonical/kubeflow-rocks/pull/39/files#diff-f44a242118806da4949b78d2e27ddaf34b56b16ba05f1f944c61548540cd9962R30)). I'd rather just leave the build step in this workflow because we'd be using the rockcraft action.

Reply:
Hmm... I need to rework `tox.ini` to have two integration test environments: one which builds ROCK and another that uses built ROCK. I will update another PR and let's discuss it.